File: packages/domain/duengung.ts
import { Entity } from "./entity";
import { Aquarium } from "./aquarium";

export interface Duengung extends Entity {
  id: string;
  menge: number;
  datum: Date;
  duenger: string;
  aquarium: Aquarium;
}

export interface DuengungFormValues {
  id?: string;
  menge: number;
  datum: Date;
  duenger: string;
  aquarium: Aquarium;
}

File: Backend/Aqua.Domain/Duengung.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace com.marcelbenders.Aqua.Domain;

public class Duengung : IEntity
{
    public Guid Id { get; set; }

    public string UserId { get; set; }

    [ForeignKey("UserId")] 
    public virtual AppUser AppUser { get; set; }

    public double Menge { get; set; }

    public DateTimeOffset Datum { get; set; }

    public string Duenger { get; set; }

    public Guid AquariumId { get; set; }
    [ForeignKey("AquariumId")]
    public virtual Aquarium Aquarium { get; set; }
}